platform/ios/app/MBXStateManager.m

```diff
@@ -0,0 +1,108 @@
+#import "MBXStateManager.h"
+#import <Mapbox/Mapbox.h>
+#import "MBXState.h"
+#import "MBXViewController.h"
+
+@interface MBXStateManager()
+
+@property (strong, nonatomic) MBXState *currentState;
+
+@end
+
+@implementation MBXStateManager
+
++ (instancetype) sharedManager {
+    static dispatch_once_t once;
+    static MBXStateManager* sharedManager;
+    dispatch_once(&once, ^{
+        sharedManager = [[self alloc] init];
+    });
+
+    return sharedManager;
+}
+
+- (MBXState*)currentState {
+    NSMutableDictionary *mapStateDictionary = [[NSUserDefaults standardUserDefaults] objectForKey:@"mapStateKey"];
+
+
+    if (mapStateDictionary == nil) {
+        _currentState = nil;
+    } else {
+        _currentState = [[MBXState alloc] init];
+
+        if (mapStateDictionary[MBXCamera]) {
+            MGLMapCamera *unpackedCamera = [NSKeyedUnarchiver unarchiveObjectWithData:mapStateDictionary[MBXCamera]];
+            _currentState.camera = unpackedCamera;
+        }
+
+        if (mapStateDictionary[MBXShowsUserLocation]) {
```

Comment:
Aside from checking if the camera exists, I think we could remove all the `if`s clauses and assign directly the values.